test/svid/jwtsvid/test_jwt_svid.py

```diff
@@ -0,0 +1,278 @@
+                ),
+                'aud': ['something'],
+                'sub': 'spiffeid://somwhere.over.the',
+            },
+            ['something'],
+        ),
+    ],
+)
+def test_valid_input_validate_claims(test_input_payload, test_input_audience):
+    JwtSvid._validate_claims(test_input_payload, test_input_audience)
+    assert True
+
+
+"""
+    parse_insecure tests
+"""
+
+
+@pytest.mark.parametrize(
+    'test_input_token,test_input_audience, expected',
+    [
+        ('', [], EMPTY_TOKEN_ERROR),
+        ('', None, EMPTY_TOKEN_ERROR),
+        (None, [], EMPTY_TOKEN_ERROR),
+        (None, None, EMPTY_TOKEN_ERROR),
+    ],
+)
+def test_invalid_input_parse_insecure(test_input_token, test_input_audience, expected):
+    with pytest.raises(ValueError) as exception:
+        JwtSvid.parse_insecure(test_input_token, test_input_audience)
+
+    assert str(exception.value) == expected
+
+
+@pytest.mark.parametrize(
+    'test_input_token,test_input_audience, expected',
+    [
+        (
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJzcGlmZmVJRDovL3Rlcy5kb21haW4vIiwibmFtZSI6IkdsYXVjaW1hciBBZ3VpYXIiLCJpYXQiOjE1MTYyMzkwMjJ9.DZhQWvCRCY96yXJzRUMiSnB6mlMUQW4il0UQ4LXAKlU",
+            [],
+            MISSING_X_ERROR.format('aud'),
+        ),  # no aud
+        (
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJzcGlmZmVJRDovL3Rlcy5kb21haW4vIiwibmFtZSI6IkdsYXVjaW1hciBBZ3VpYXIiLCJpYXQiOjE1MTYyMzkwMjIsImF1ZCI6InNwaXJlIn0.hL6rIVn5dFuQ4KKWxZ7yag7Gi68m174RqaU04720PZU",
+            ["spire"],
+            MISSING_X_ERROR.format('exp'),
+        ),  # no exp
+        (
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiR2xhdSIsImlhdCI6MTUxNjIzOTAyMiwiZXhwIjoxNTE2MjM5MDIyLCJhdWQiOlsic3BpcmUiXX0.Cc6vnXybg_7DTYpObYkCNIuJqzROlRj3jVe0g4qO-W4",
+            ["spire"],
+            MISSING_X_ERROR.format('sub'),
+        ),  # no sub
+        (
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJzcGlmZmVJRDovL3Rlcy5kb21haW4vIiwibmFtZSI6IkdsYXVjaW1hciBBZ3VpYXIiLCJpYXQiOjE1MTYyMzkwMjIsImF1ZCI6InNwaXJlIiwiZXhwIjoiMTU0NTE4NTQ5NiJ9.HQ6F_uvq597L1TunY6RSe0OpOAF-r2vAVGIFDQrde1c",
+            ["spire"],
+            SIGNATURE_EXPIRED_ERROR,
+        ),
+    ],
+)
+def test_invalid_parse_insecure(test_input_token, test_input_audience, expected):
+    with pytest.raises(JwtSvidError) as exception:
+        JwtSvid.parse_insecure(test_input_token, test_input_audience)
+
+    assert str(exception.value) == expected
+
+
+@pytest.mark.parametrize(
+    'test_input_token,test_input_audience, expected',
+    [
+        (
+            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJzcGlmZmU6Ly90ZXN0Lm9yZy8iLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjIsImV4cCI6NDcyMTUwNTQzNywiYXVkIjpbInNwaXJlIl19.cvoi48geyYPAP88RQbBXyEXoRX09II5xAR0XKULWDwQ",
+            ["spire"],
+            "spiffe://test.org/",
+        ),
+    ],
+)
+def test_valid_parse_insecure(test_input_token, test_input_audience, expected):
```

Comment:
add another test case, with a different token and multiple audiences

Reply:
I will do for _validate_claims not for parse_insecure, I was expecting the discussion about the right way to validate it. Thanks for the reminder.